src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -616,29 +626,127 @@ export class CPlusPlusRenderer extends ConvenienceRenderer {
+                        this.emitLine("void ", setterName, "(", rendered, " value) { if (value) checkConstraint(\"", name, "\", ", name, "Constraint, *value); this->", name, " = value; }");   
+                    } else {
+                        this.emitLine("void ", setterName, "(", rendered, " value) { this->", name, " = value; }");   
+                    }
                 } else {
                     this.emitLine("const ", rendered, " & ", getterName, "() const { return ", name, "; }");
                     this.emitLine(rendered, " & ", mutableGetterName, "() { return ", name, "; }");
-                    this.emitLine("void ", setterName, "(const ", rendered, "& value) { this->", name, " = value; }");
+                    if (constraints !== undefined && constraints.has(jsonName)) {
+                        this.emitLine("void ", setterName, "(const ", rendered, "& value) { checkConstraint(\"", name, "\", ", name, "Constraint, value); this->", name, " = value; }");   
+                    } else {
+                        this.emitLine("void ", setterName, "(const ", rendered, "& value) { this->", name, " = value; }");   
+                    }
                 }
                 this.ensureBlankLine();
             }
         });
     }
 
+    protected generateClassConstraints(c: ClassType): Map<string, string> | undefined {
+        let res: Map<string,string> = new Map<string, string>();
+        this.forEachClassProperty(c, "none", (_name, jsonName, property) => {
+            let constrArg:string="(";
+            const minmaxval = minMaxValue(property.type);
+            constrArg += minmaxval !== undefined && minmaxval[0] !== undefined ? minmaxval[0] : "boost::none";
+            constrArg += ", ";
+            constrArg += minmaxval !== undefined && minmaxval[1] !== undefined ? minmaxval[1] : "boost::none";
+            constrArg += ", ";
+            const minmaxlen = minMaxLength(property.type);
+            constrArg += minmaxlen !== undefined && minmaxlen[0] !== undefined ? minmaxlen[0] : "boost::none";
+            constrArg += ", ";
+            constrArg += minmaxlen !== undefined && minmaxlen[1] !== undefined ? minmaxlen[1] : "boost::none";
+            constrArg += ", ";
+            const patt = pattern(property.type);
+            constrArg += patt === undefined ? "boost::none" : "std::string(\""+patt+"\")";
+            constrArg += ")";
+
+            if (minmaxval !== undefined ||
+                minmaxlen !== undefined ||
+                patt !== undefined) {
+                res.set(jsonName, jsonName+"Constraint"+constrArg);
+            }
+        });
+
+        return res.size === 0 ? undefined : res;
+    }
+
+    protected emitStringGenerator(c: ClassType): void {
+        this.forEachClassProperty(c, "none", (name, _jsonName, property) => {
+            const [getterName, , ] = defined(this._gettersAndSettersForPropertyName.get(name));
+            if (property.type.kind === "array") {
```

Comment:
What if the property is an array of maps of enums?

Reply:
I'm fine if the code doesn't handle that case fully, but it shouldn't emit invalid code.